dc/s2n-quic-dc/src/sync/mpsc.rs

```diff
@@ -114,6 +121,13 @@ impl<T> Drop for Receiver<T> {
 }
 
 impl<T> Receiver<T> {
+    #[inline]
+    pub fn sender(&self) -> Sender<T> {
+        Sender {
+            channel: self.channel.clone_for_sender(),
```

Comment:
This seems a bit odd -- we close the channel when the sender count drops to zero, but this would allow "reopening" the sender side. Maybe needs an assert!(...) that the sender count isn't zero?

Reply:
Yes it's a bit goofy; it's mostly the way it is to work around the current impl where we create the sender after initializing the descriptor, which would start out with `0` senders:

```rust
                let descriptor = Descriptor::new(descriptor);
                let sender = Sender::new(descriptor.clone_for_sender());
```

Like you said, it's an internal implementation detail so I can revisit later.